### PR TITLE
[RAPTOR-16922] add dr workload code versions command

### PR DESCRIPTION
**Pull request by wojtekwdr**
# RATIONALE

`dr workload code versions` command lists the catalog version history for the workload artifact that the current project directory is linked to.

The command depends on the `filesapi` package, which lives only on `wojtekw/RAPTOR-16924-foundation`, so this branch is stacked on top of foundation. Once foundation lands, this branch can be re-targeted to main with a simple rebase.

### filesapi

- New `CatalogVersion` and `CatalogVersionsResp` types in `types.go`, with JSON tags matching the real wire shape (`catalogVersionId`, `creationDate`, `numFiles`, `size`).
- New `ListVersions(catalogID, limit)` method on `filesapi.Client`. Paginates through `Next` and reuses the cross-host safety check from `AllFiles`.
- A regression test pinning the JSON field names, so future renames cannot silently break decoding.

### cmd

- New `cmd/workload/code/versions/` package, registered alongside `init`.
- Auth gated through `PreRunE: auth.EnsureAuthenticatedE`, same as the other workload commands.
- Test seam vars (`getArtifactFn`, `newClientFn`) follow the precedent in `code/init`.
- 404 from the artifact GET or the version listing is specialized to `artifact X not found` and `catalog X not found`, mirroring init.
- `--limit` rejects negative values, and 0 means unlimited (per the spec).

### Display

Human output is a tab-aligned table with a `*` marker on the current `codeRef` row, and a trailing line that names the locally synced version. JSON output is a flat array of objects with `versionId`, `versionShort`, `createdAt`, `fileCount`, `totalSize`, and `isCurrent`.

## TESTING

Unit tests cover filesapi pagination, the limit cap, cross-host rejection, real-wire-shape decoding, and the empty case. Cmd tests cover not-linked, no-catalog, text and JSON output, the negative-limit rejection, and both 404 paths. Manually smoke-tested against staging on an artifact with four real versions.

## NOTES

Stacked on `wojtekw/RAPTOR-16924-foundation` (PR 470). Review foundation first. No changes outside the listed files.

---

> [!NOTE]
> **Medium Risk**
> Adds a new CLI subcommand and a new Files API client method with pagination, which could affect users via incorrect listings or API error handling. Security risk is low, but it introduces new network calls and output formatting paths that should be validated against real server responses.
> 
> **Overview**
> Adds `dr workload code versions` under `dr workload code`, which reads the local `.wapi/` link state, fetches the linked artifact and its Files API catalog version history, and renders either a table or `--output-format json` output marking the artifact’s current `codeRef` version and the locally last-synced version.
> 
> Extends `internal/drapi/filesapi` with `ListVersions(catalogID, limit)` plus new response/types for decoding the versions endpoint, including cursor pagination with same-host enforcement. Includes unit tests for the command behavior (not-linked/no-catalog, limit validation, JSON/text output, 404 specialization) and for the Files API pagination/decoding edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 686930dddf7336077eac37e0d4bd73b7a53cfa2a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by ajalon1 — CHANGES_REQUESTED** [review]
One note here, and I'm sorry I forgot to mention this earlier, but we use `bubbletea` for rendering human-readable tables.

Examples:

image

image


I'll let Claude explain:
> Before merging, refactor renderText() in cmd/workload/code/versions/view.go to use charmbracelet/lipgloss/table:

> Import lipgloss, lipgloss/table, and use tui.TableBorderStyle
> Build table with .Headers(), .Row() for each version
Use tui styles for column styling (similar to plugin/list)
Keep marker logic (maybe as a prefix in first column or suffix column)
> Update text output tests to verify actual rendered output
This will take ~30 minutes and ensure visual consistency across the CLI.

Otherwise this LGTM.

File: cmd/workload/code/versions/cmd.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/datarobot/cli/internal/workload"
+	"github.com/datarobot/cli/internal/workload/wapi"
+	"github.com/spf13/cobra"
+)
+
+// Test seams.
+var (
+	getArtifactFn = workload.GetArtifact
+	newClientFn   = filesapi.New
+)
```
Comment (ajalon1) [inline cmd/workload/code/versions/cmd.go]: Again, use the deps dependency inversion pattern as described in #477`. You're almost there. :)

File: cmd/workload/code/versions/cmd.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  dr workload code versions --limit 10
+  dr workload code versions --output-format json`,
+		PreRunE: auth.EnsureAuthenticatedE,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runVersions(cmd, outputFormat)
+		},
+	}
+
+	c.Flags().String("dir", "", "Project directory (default: current directory).")
+	c.Flags().Int("limit", 0, "Maximum number of versions to show (0 = all).")
```
Comment (ajalon1) [inline cmd/workload/code/versions/cmd.go]: How many could you have? Could all = 100? 1000? 1000000?

File: cmd/workload/code/versions/cmd.go (hunk 1 above)
Comment (wojtekwdr) [reply cmd/workload/code/versions/cmd.go]: refactored according to description

File: cmd/workload/code/versions/cmd.go (hunk 2 above)
Comment (wojtekwdr) [reply cmd/workload/code/versions/cmd.go]: I aligned it with `dr workload artifact list` where `--limit 0` throws error

File: cmd/workload/code/versions/cmd.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+func defaultDeps() Deps {
+	return Deps{
+		GetArtifact: workload.GetArtifact,
+		Files:       filesapi.New(),
+	}
+}
+
+func Cmd() *cobra.Command {
+	return cmdWithDeps(defaultDeps())
```
[the remark this answers is not in the document]
Comment (ajalon1) [reply cmd/workload/code/versions/cmd.go:48]: You can add telemetry in a follow-up; that is an incomplete feature at this point.

**Review by ajalon1 — APPROVED** [review]
